src/reader/mod.rs

```diff
@@ -1020,15 +1020,15 @@ enum BangType {
     /// <!--...-->
     Comment,
     /// <!DOCTYPE...>
-    DocType,
+    DocType(i32),
```

Comment:
Updating documentation also would be useful (language can be improved if required, I'm not native speaker):
```rust
    /// <!DOCTYPE...>. Contains a balance of `<` (+1) and `>` (-1)
    DocType(i32),
```